src/main/java/com/iexec/common/docker/client/DockerClientInstance.java

```diff
@@ -557,8 +554,11 @@ public HostConfig buildHostConfigFromRunRequest(DockerRunRequest dockerRunReques
             hostConfig.withBinds(Binds.fromPrimitive(
                     dockerRunRequest.getBinds().toArray(new String[0])));
         }
-        if (dockerRunRequest.getDevices() != null) {
-            hostConfig.withDevices(dockerRunRequest.getDevices());
+        try {
+            List<Device> devices = new ArrayList<>(dockerRunRequest.getDevices());
+            hostConfig.withDevices(devices);
+        } catch (NullPointerException e) {
+            log.info("No devices defined in docker run request.");
```

Comment:
Why not providing a default empty list instead of risking some NPE later?

Reply:
This is the only way to perform only one call to `dockerRunRequest.getDevices()`.
All other implementations will require a check on `dockerRunRequest.getDevices() != null` then `addAll(dockerRunRequest.getDevices())`.
If the goal is to avoid multiple calls to `getDevices`, we *have to* access it once and handle the `NullPointerException` if it happens.

Reply:
Implementation without NullPointerException in https://github.com/iExecBlockchainComputing/iexec-common/pull/296/commits/e05ddc496cd2f06a4a5cdd9cd3ec8422b575dea8